bin/datatool/src/util.rs

```diff
@@ -0,0 +1,417 @@
+
+            Some(buf)
+        }
+        None => None,
+    };
+
+    // Parse each of the operator keys.
+    let mut opkeys = Vec::new();
+
+    if let Some(opkeys_path) = cmd.opkeys {
+        let opkeys_str = fs::read_to_string(opkeys_path)?;
+
+        for l in opkeys_str.lines() {
+            // skip lines that are empty or look like comments
+            if l.trim().is_empty() || l.starts_with("#") {
+                continue;
+            }
+
+            opkeys.push(parse_xpub(l)?);
+        }
+    }
+
+    for k in cmd.opkey {
+        opkeys.push(parse_xpub(&k)?);
+    }
+
+    // Parse the deposit size str.
+    let deposit_sats = cmd
+        .deposit_sats
+        .map(|s| parse_abbr_amt(&s))
+        .transpose()?
+        .unwrap_or(1_000_000_000);
+
+    // Parse the checkpoint verification key.
+    let rollup_vk: Buf32 = GUEST_CHECKPOINT_VK_HASH_STR
+        .parse()
+        .expect("invalid checkpoint verifier key hash");
+
+    let config = ParamsConfig {
+        name: cmd.name.unwrap_or_else(|| "strata-testnet".to_string()),
+        bitcoin_network: ctx.bitcoin_network,
+        // TODO make these consts
+        block_time_sec: cmd.block_time.unwrap_or(15),
+        epoch_slots: cmd.epoch_slots.unwrap_or(64),
+        genesis_trigger: cmd.genesis_trigger_height.unwrap_or(100),
+        seqkey,
+        opkeys,
+        rollup_vk,
+        // TODO make a const
+        deposit_sats,
+        proof_timeout: cmd.proof_timeout,
+    };
+
+    let params = construct_params(config);
+    let params_buf = serde_json::to_string_pretty(&params)?;
+
+    if let Some(out_path) = &cmd.output {
+        fs::write(out_path, params_buf)?;
+        eprintln!("wrote to file {out_path:?}");
+    } else {
+        println!("{params_buf}");
+    }
+
+    Ok(())
+}
+
+/// Generates a new [`Xpriv`] that will [`Zeroize`](zeroize) on [`Drop`].
+///
+/// # Notes
+///
+/// Takes a mutable reference to an RNG to allow flexibility in testing.
+/// The actual generation requires a high-entropy source like [`OsRng`](rand::rngs::OsRng)
+/// to securely generate extended private keys.
+fn gen_priv<R: CryptoRng + RngCore>(rng: &mut R, net: Network) -> ZeroizableXpriv {
+    let mut seed = [0u8; 32];
+    rng.fill_bytes(&mut seed);
+    Xpriv::new_master(net, &seed).expect("valid seed").into()
+}
+
+/// Reads an [`Xpriv`] from file as a string and verifies the checksum.
+///
+/// # Notes
+///
+/// This [`Xpriv`] will [`Zeroize`](zeroize) on [`Drop`].
+fn read_xpriv(path: &Path) -> anyhow::Result<ZeroizableXpriv> {
+    let raw_buf = fs::read(path)?;
+    let str_buf = std::str::from_utf8(&raw_buf)?;
+    let buf = base58::decode_check(str_buf)?;
+    Ok(Xpriv::decode(&buf)?.into())
+}
+
+/// Parses an [`Xpriv`] from environment variable.
+///
+/// # Notes
+///
+/// This [`Xpriv`] will [`Zeroize`](zeroize) on [`Drop`].
+fn parse_xpriv_from_env(env: &'static str) -> anyhow::Result<Option<ZeroizableXpriv>> {
+    let Ok(val) = std::env::var(env) else {
```

Comment:
This is a secret value that is not zeroized.

Reply:
Done in 965aaee1b7b4318eb8a72b0a873ab5592b15bde6